python_modules/dagster/dagster_tests/execution_tests/engine_tests/test_step_deletegating_executor_retries.py

```diff
@@ -0,0 +1,145 @@
+
+    def launch_step(self, step_handler_context):
+
+        assert step_handler_context.execute_step_args.step_keys_to_execute == ["retry_op"]
+
+        attempt_count = (
+            step_handler_context.execute_step_args.known_state.get_retry_state().get_attempt_count(
+                "retry_op"
+            )
+        )
+        if attempt_count == 0:
+            assert TestStepHandler.launched_first_attempt is False
+            assert TestStepHandler.launched_second_attempt is False
+            TestStepHandler.launched_first_attempt = True
+        elif attempt_count == 1:
+            assert TestStepHandler.launched_first_attempt is True
+            assert TestStepHandler.launched_second_attempt is False
+            TestStepHandler.launched_second_attempt = True
+        else:
+            raise Exception("Unexpected attempt count")
+
+        print("TestStepHandler Launching Step!")  # pylint: disable=print-call
+        TestStepHandler.processes.append(
+            subprocess.Popen(step_handler_context.execute_step_args.get_command_args())
+        )
+        return iter(())
+
+    def check_step_health(self, step_handler_context) -> CheckStepHealthResult:
+
+        assert step_handler_context.execute_step_args.step_keys_to_execute == ["retry_op"]
+
+        attempt_count = (
+            step_handler_context.execute_step_args.known_state.get_retry_state().get_attempt_count(
+                "retry_op"
+            )
+        )
+        if attempt_count == 0:
+            assert TestStepHandler.launched_first_attempt is True
+            assert TestStepHandler.launched_second_attempt is False
+        elif attempt_count == 1:
+            assert TestStepHandler.launched_first_attempt is True
+            assert (
+                TestStepHandler.launched_second_attempt
+            ), "Second attempt not launched, shouldn't be checking on it"
+
+        return CheckStepHealthResult.healthy()
+
+    def terminate_step(self, step_handler_context):
+        raise NotImplementedError()
+
+    @classmethod
+    def reset(cls):
+        cls.launched_first_attempt = False
+        cls.launched_second_attempt = False
+
+    @classmethod
+    def wait_for_processes(cls):
+        for p in cls.processes:
+            p.wait(timeout=5)
+
+
+@executor(
+    name="retry_assertion_executor",
+    requirements=multiple_process_executor_requirements(),
+    config_schema=Permissive(),
+)
+def retry_assertion_executor(exc_init):
+    return StepDelegatingExecutor(
+        TestStepHandler(),
+        **(merge_dicts({"retries": RetryMode.ENABLED}, exc_init.executor_config)),
+        check_step_health_interval_seconds=0,
+    )
+
+
+@op(config_schema={"fails_before_pass": int})
+def retry_op(context: OpExecutionContext):
+    if context.retry_number < context.op_config["fails_before_pass"]:
+        # enough for check_step_health to be called, since we set check_step_health_interval_seconds=0
+        raise RetryRequested(seconds_to_wait=5)
```

Comment:
you can set `max_retries` on here and have it exhaust that way

Reply:
true, I think the default of 1 is fine though